pkg/server/plugin/upstreamca/awssecret/awslib.go

```diff
@@ -40,6 +43,19 @@ func newSecretsManagerClient(config *AWSSecretConfiguration, region string) (sec
 		awsConfig.Credentials = credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, config.SecurityToken)
 	}
 
+	// Optional: Assuming role
+	if config.AssumeRoleARN != "" {
+		staticsess, err := session.NewSession(&aws.Config{Credentials: awsConfig.Credentials})
+		if err != nil {
+			return nil, err
+		}
+		awsConfig.Credentials = credentials.NewCredentials(&stscreds.AssumeRoleProvider{
+			Client:   sts.New(staticsess),
+			RoleARN:  config.AssumeRoleARN,
+			Duration: 30 * time.Minute,
```

Comment:
What impact does this duration have? Will a new client be created on each rotation making this duration the max lifetime of a single operation?

Reply:
Yes, it's a cutoff for how long these credentials will be valid. The default is 15 minutes and we could actually make it shorter considering (changed it to 5). This will only be used once to fetch.

Reply:
https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/#pkg-variables